### PR TITLE
feat: support latest binary alert feature through cli 

### PR DESCRIPTION
**Pull request by scortier**
### Description
If the user wants to check whether any latest binary release is available (it can be minor release too).

**Just need to trigger the command to check the released update : `optimus version`**

### Preview

Screenshot 2021-07-28 at 10 16 19 AM

### Related Issues 
Issue #27

### CODE REVIEW

**Comment by scortier** [comment]
Please review the PR @kushsharma

**Review by kushsharma — CHANGES_REQUESTED** [review]

File: cmd/version.go [hunk 1]
```diff
@@ -1,10 +1,18 @@
+//version.go
```
Comment (kushsharma) [inline cmd/version.go:1]: Why is this here?

File: cmd/version.go [hunk 2]
```diff
@@ -15,8 +21,45 @@ import (
...
 )
 
+func checkLatestVersion() string {
+	gitClient := http.Client{
+		Timeout: clientVersion,
+	}
+
+	req, err := http.NewRequest(http.MethodGet, apiURL, nil)
+	if err != nil {
+		log.Warn("failed to create request for latest version")
```
Comment (kushsharma) [inline cmd/version.go:35]: If an error occurs it should not continue to execute the function and return an empty string

File: cmd/version.go [hunk 3]
```diff
@@ -15,8 +21,45 @@ import (
 
 const (
 	versionTimeout = time.Second * 2
+	clientVersion  = time.Second * 1
+	apiURL         = "https://api.github.com/repos/odpf/optimus/releases/latest"
 )
 
+func checkLatestVersion() string {
```
Comment (kushsharma) [inline cmd/version.go:28]: lets move this function after `func versionCommand`

File: cmd/version.go [hunk 4]
```diff
@@ -25,6 +68,18 @@ func versionCommand(l logger, host string) *cli.Command {
...
+			currentV, err := version.NewVersion(config.Version)
+			if err != nil {
+				return errors.Wrapf(err, "failed to parse current version")
+			}
+			latestV, err := version.NewVersion(checkLatestVersion())
+			if err != nil {
+				return errors.Wrapf(err, "failed to parse latest version")
+			}
+			if currentV.LessThan(latestV) {
+				l.Printf(fmt.Sprintf("new version is available : %s , consider updating the client", coloredNotice(latestV)))
```
Comment (kushsharma) [inline cmd/version.go:80]: why space before comma?

File: cmd/version.go [hunk 5]
```diff
@@ -25,6 +68,18 @@ func versionCommand(l logger, host string) *cli.Command {
 		RunE: func(c *cli.Command, args []string) error {
 			l.Printf(fmt.Sprintf("client: %s-%s", coloredNotice(config.Version), config.BuildCommit))
 
+			currentV, err := version.NewVersion(config.Version)
+			if err != nil {
+				return errors.Wrapf(err, "failed to parse current version")
+			}
+			latestV, err := version.NewVersion(checkLatestVersion())
+			if err != nil {
+				return errors.Wrapf(err, "failed to parse latest version")
```
Comment (kushsharma) [inline cmd/version.go:77]: when `checkLatestVersion` will return an empty string at the time of error, we should not return error here, simply ignore. We want this check for update flow non-obstructive

File: go.mod [hunk 6]
```diff
@@ -17,12 +15,14 @@ require (
 	github.com/golang/protobuf v1.5.2
 	github.com/google/uuid v1.1.2
 	github.com/googleapis/google-cloud-go-testing v0.0.0-20200911160855-bcd43fbb19e8
+	github.com/gorilla/mux v1.7.4
```
Comment (kushsharma) [inline go.mod:18]: I don't think we use mux anywhere, can you consider running `go mod tidy`?

**Review by kushsharma — CHANGES_REQUESTED** [review]

File: cmd/version.go (hunk 2 above)
Comment (scortier) [reply cmd/version.go:35]: But i think we discussed, that the execution of the command should not stop regardless of the error , we will just warn the user about the error

Comment (kushsharma) [reply cmd/version.go:69]: we are not stopping the execution of the command but how do you expect this function to work if it failed to create even a valid http request.

File: cmd/version.go [hunk 7]
```diff
@@ -27,6 +34,18 @@ func versionCommand(l logger, host string, pluginRepo models.PluginRepository) *
...
+			}
+
+			if currentV.LessThan(latestV) {
+				l.Printf(fmt.Sprintf("new version is available : %s, consider updating the client", coloredNotice(latestV)))
+			}
```
Comment (kushsharma) [inline cmd/version.go:48]: what if we move all this to `checkLatestVersion` and take `config.Version` as an argument? This will avoid failing for errors when parsing the latest version at line number 43.

**Review by kushsharma — CHANGES_REQUESTED** [review]
